jerry-main/fuzzilli.c

```diff
@@ -0,0 +1,300 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "jerryscript.h"
+#include "jerryscript-ext/handler.h"
+
+/**
+ * Maximum size of source code
+ */
+#define JERRY_BUFFER_SIZE (1048576)
+static uint8_t buffer[JERRY_BUFFER_SIZE];
+
+#include <stdio.h>
+#include <assert.h>
+#include <stdlib.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <unistd.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <sys/wait.h>
+#include <sys/mman.h>
+
+/**
+ * Fuzzilli interface, for more information refer to:
+ * https://github.com/googleprojectzero/fuzzilli/tree/master/Targets
+ * https://github.com/googleprojectzero/fuzzilli/blob/master/Targets/coverage.c
+ */
+
+// BEGIN FUZZING CODE
```

Comment:
Alright, NOW I see what's happening here... So, this is something that is actually in `fuzzilli/Targets/coverage.c`. That code has a quite different coding style. If we applied our coding style and additional changes to that code, it could be a nightmare to sync/maintain it in the future. So, why don't we just add that file to `third-party/fuzzilli/coverge.c` and include(!) "coverage.c" from here. Then maintenance, if ever needed, becomes just copying that file from that project to this one.

Something like the following might be needed in `jerry-main/CMakeLists.txt`:
```cmake
target_include_directories("jerry-fuzzilli" PRIVATE "${CMAKE_SOURCE_DIR}/third-party/fuzzilli")
```